### ISSUE TITLE
About the large dataset and Unet Training

### ISSUE THREAD

**Issue by liuyuemaicha** [kvablack/ddpo-pytorch#7]
Thanks for your great work!

I have repeated the lora training，and got a same results in supplied prompt.  But I have some issues in large dataset and Unet Trainning.
##
## Problem1: Overfitting
When I training the LoRA with 400 prompts, I find the reward overfitting to maximum easily in 4K steps.  Even I update the prompts to 20K, the same problem will happen.
So, is it work for large dataset ? How to train the ddpo in a large dataset ?
### 400 prompts
![image](https://github.com/kvablack/ddpo-pytorch/assets/17443897/d2491b6b-8f78-4c64-b25a-61a49f555b90)
### 20K prompts
![image](https://github.com/kvablack/ddpo-pytorch/assets/17443897/59aeb86a-d976-43d1-9a8c-c4cf15156d03)

## Problem2: Unet Trainning
I set "config.use_lora = False" in config/base.py to train Unet, but the reward change to zero within tens steps, and the sd model generate black images.

![image](https://github.com/kvablack/ddpo-pytorch/assets/17443897/fb755cf7-6c49-473b-8c77-d6ff63abf592)

**Comment by liuyuemaicha** [comment]
Thanks for your great work!

I have repeated the lora training，and got a same results in supplied prompt.  But I have some issues in large dataset and Unet Trainning.

1 . Overfitting
When I training the LoRA with 400 prompts, I find the reward overfitting to maximum(one) easily in 4K steps.  Even I update the prompts to 20K, the same problem will happen.
So, is it work for large dataset ? How to train the ddpo in the large dataset ?

2. Unet Trainning
I set "config.use_lora = False" in config/base.py to train Unet, but the reward change to zero within tens steps, and the sd model generate black images.
![image](https://github.com/kvablack/ddpo-pytorch/assets/17443897/fb755cf7-6c49-473b-8c77-d6ff63abf592)

**Comment by desaixie** [comment]
Same problem here. Training with LoRA disabled, the network starts to produce broken images after 1 epoch and gets worse as training continues. This happens with compressibility reward and with my custom prompt and reward functions and a different diffusion model.
![image](https://github.com/kvablack/ddpo-pytorch/assets/32966374/63886b67-920f-42ea-b32c-d89e2d8bb8cf)
![image](https://github.com/kvablack/ddpo-pytorch/assets/32966374/391b836b-4d61-4164-9e85-60c4065659b0)
![image](https://github.com/kvablack/ddpo-pytorch/assets/32966374/4457aedd-160f-4e30-87b0-080e3c2db92d)
Above are epochs 0 (before training starts), 1, and 2 when trained with compressibility.

**Comment by kvablack** [comment]
1. For such a large dataset, you may need some more advanced techniques, such as a prompt-dependent value function.
2. I would recommend a lower learning rate for full finetuning, such as 1e-5.

**Comment by liuyuemaicha** [comment]
> 1. For such a large dataset, you may need some more advanced techniques, such as a prompt-dependent value function.
> 2. I would recommend a lower learning rate for full finetuning, such as 1e-5.

Thanks for your answers. I trans the reward function to bigG and the problem solved. Thanks!
